pkg/controllers/cliconfig/cli_config_controller.go

```diff
@@ -0,0 +1,472 @@
+				SessionMaxAgeSeconds: 5 * 60, // 5 minutes
+				SessionName:          "ssn",
+			},
+		},
+	}
+
+	cliConfigBytes, err := runtime.Encode(encoder, cliConfig)
+	if err != nil {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    "OAuthConfigDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "EncodeFailed",
+				Message: fmt.Sprintf("Failed to encode CLI config: %v", err),
+			},
+		}
+	}
+
+	observedConfig, err := grabPrefixedConfig(operatorConfig.Spec.ObservedConfig.Raw, configobservation.OAuthServerConfigPrefix)
+	if err != nil {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    "OAuthConfigDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "GetOAuthServerConfigFailed",
+				Message: fmt.Sprintf("Unable to get oauth-server configuration: %v", err),
+			},
+		}
+	}
+
+	completeConfigBytes, err := resourcemerge.MergePrunedProcessConfig(&osinv1.OsinServerConfig{}, nil, cliConfigBytes, observedConfig, operatorConfig.Spec.UnsupportedConfigOverrides.Raw)
+	if err != nil {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    "OAuthConfigDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "MergeConfigFailed",
+				Message: fmt.Sprintf("Failed to merge config with unsupportedConfigOverrides: %v", err),
+			},
+		}
+	}
+
+	expectedCLIConfig := getCliConfigMap(completeConfigBytes)
+
+	_, _, err = resourceapply.ApplyConfigMap(c.configMaps, recorder, expectedCLIConfig)
+	if err != nil {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    "OAuthConfigDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "ApplyFailed",
+				Message: fmt.Sprintf("Failed to apply CLI configuration %q: %v", expectedCLIConfig.Name, err),
+			},
+		}
+	}
+
+	return nil
+}
+
+// grabPrefixedConfig returns the configuration from the operator's observedConfig field
+// in the subtree given by the prefix
+func grabPrefixedConfig(observedBytes []byte, prefix ...string) ([]byte, error) {
+	if len(prefix) == 0 {
+		return observedBytes, nil
+	}
+
+	prefixedConfig := map[string]interface{}{}
+	if err := json.NewDecoder(bytes.NewBuffer(observedBytes)).Decode(&prefixedConfig); err != nil {
+		klog.V(4).Infof("decode of existing config failed with error: %v", err)
+	}
+
+	actualConfig, _, err := unstructured.NestedFieldCopy(prefixedConfig, prefix...)
+	if err != nil {
+		return nil, err
+	}
+
+	return json.Marshal(actualConfig)
+}
+
+func getCliConfigMap(completeConfigBytes []byte) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "v4-0-config-system-cliconfig",
+			Namespace: "openshift-authentication",
+			Labels: map[string]string{
+				"app": "oauth-openshift",
+			},
+			Annotations:     map[string]string{},
+			OwnerReferences: nil, // TODO
+		},
+		Data: map[string]string{
+			"v4-0-config-system-cliconfig": string(completeConfigBytes),
+		},
+	}
+}
+
+func getMasterCA() *string {
```

Comment:
```golang
func sptr(s string) *string {
    return &s
}
```
and inline the string const above in this function to make this all more readable